drivers/serial/serial.c

```diff
@@ -901,6 +901,11 @@ static ssize_t uart_readv(FAR struct file *filep, FAR struct uio *uio)
 #ifdef CONFIG_SERIAL_IFLOWCONTROL_WATERMARKS
   unsigned int nbuffered;
   unsigned int watermark;
+#  ifndef CONFIG_ARCH_LD_16BIT_NOT_ATOMIC
+  int16_t head;
```

Comment:
why not merge to previous patch

Reply:
Because it fixes different bug (on all architectures.) I will reorder 
the patches to make that more obvious and to have the less intrusive one 
be the first in case it turns out that the second one needs to be 
reverted.